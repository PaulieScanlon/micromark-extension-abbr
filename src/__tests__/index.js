import dedent from 'dedent';

import { micromark } from 'micromark';
import { directive, directiveHtml } from 'micromark-extension-directive';

import micromarkAbbr from '../../src/index';

it('no-config renders references', () => {
  const contents = micromarkAbbr(
    dedent`
      This is an abbreviation: HTML.

      *[HTML]: HyperText Markup Language
    `
  );

  expect(contents).toMatchSnapshot();
});

it('empty object renders references', () => {
  const contents = micromarkAbbr(
    dedent`
      This is an abbreviation: HTML.

      *[HTML]: HyperText Markup Language
    `,
    {}
  );

  expect(contents).toMatchSnapshot();
});

it('expandFirst renders references', () => {
  const contents = micromarkAbbr(
    dedent`
      This is an abbreviation: HTML.

      *[HTML]: HyperText Markup Language
    `,
    {
      expandFirst: true
    }
  );

  expect(contents).toMatchSnapshot();
});
