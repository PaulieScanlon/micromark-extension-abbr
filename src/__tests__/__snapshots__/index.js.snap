// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`empty object renders references 1`] = `
"<p>This is an abbreviation: <abbr title=\\"HyperText Markup Language\\">HTML</abbr>.</p>
"
`;

exports[`expandFirst renders references 1`] = `
"<p>This is an abbreviation: <abbr title=\\"HyperText Markup Language\\">HTML</abbr>.</p>
"
`;

exports[`no-config renders references 1`] = `
"<p>This is an abbreviation: <abbr title=\\"HyperText Markup Language\\">HTML</abbr>.</p>
"
`;
